src/core/cgroup.c

```diff
@@ -2041,60 +1747,21 @@ static void cgroup_context_apply(
-         * level. The quota exists there too, but any attempt to write to it is refused with EINVAL. Inside of
-         * containers we want to leave control of these to the container manager (and if cgroup v2 delegation is used
-         * we couldn't even write to them if we wanted to). */
+        /* In fully unified mode these attributes don't exist on the host cgroup root. */
         if ((apply_mask & CGROUP_MASK_CPU) && !is_local_root) {
+                uint64_t weight;
 
-                if (cg_all_unified() > 0) {
-                        uint64_t weight;
-
-                        if (cgroup_context_has_cpu_weight(c))
-                                weight = cgroup_context_cpu_weight(c, state);
-                        else if (cgroup_context_has_cpu_shares(c)) {
-                                uint64_t shares;
-
-                                shares = cgroup_context_cpu_shares(c, state);
-                                weight = cgroup_cpu_shares_to_weight(shares);
-
-                                log_cgroup_compat(u, "Applying [Startup]CPUShares=%" PRIu64 " as [Startup]CPUWeight=%" PRIu64 " on %s",
-                                                  shares, weight, path);
-                        } else
-                                weight = CGROUP_WEIGHT_DEFAULT;
-
-                        cgroup_apply_unified_cpu_idle(u, weight);
-                        cgroup_apply_unified_cpu_weight(u, weight);
-                        cgroup_apply_unified_cpu_quota(u, c->cpu_quota_per_sec_usec, c->cpu_quota_period_usec);
-
-                } else {
-                        uint64_t shares;
-
-                        if (cgroup_context_has_cpu_weight(c)) {
-                                uint64_t weight;
-
-                                weight = cgroup_context_cpu_weight(c, state);
-                                shares = cgroup_cpu_weight_to_shares(weight);
-
-                                log_cgroup_compat(u, "Applying [Startup]CPUWeight=%" PRIu64 " as [Startup]CPUShares=%" PRIu64 " on %s",
-                                                  weight, shares, path);
-                        } else if (cgroup_context_has_cpu_shares(c))
-                                shares = cgroup_context_cpu_shares(c, state);
-                        else
-                                shares = CGROUP_CPU_SHARES_DEFAULT;
+                if (cgroup_context_has_cpu_weight(c))
+                        weight = cgroup_context_cpu_weight(c, state);
+                else
+                        weight = CGROUP_WEIGHT_DEFAULT;
 
-                        cgroup_apply_legacy_cpu_shares(u, shares);
-                        cgroup_apply_legacy_cpu_quota(u, c->cpu_quota_per_sec_usec, c->cpu_quota_period_usec);
-                }
+                cgroup_apply_unified_cpu_idle(u, weight);
+                cgroup_apply_unified_cpu_weight(u, weight);
+                cgroup_apply_unified_cpu_quota(u, c->cpu_quota_per_sec_usec, c->cpu_quota_period_usec);
```

Comment:
I'd add a TODO item of removing unified from function names once we're legacy-free internally